docs/model/model-rules.md

```diff
@@ -0,0 +1,53 @@
+# Model Rules
+In this article you will learn about model rules, what is and what is not model, and how to correctly create and insert new functionality into existing directory architecture.
+
+## Hierarchy nesting
+All classes related to a model unit are nested into `\Shopsys\FrameworkBundle\Model\<MODEL>`.
+
+They can also be nested into deeper directory such as `Shopsys\FrameworkBundle\Model\Product\Search` if it encapsulate group of classes representing some specific functionality, for example search functionality for a Product entity.
+
+## Rules
+- classes always must be in right FQN, if you want to add a new functionality that works with product, your class should be created into `\Shopsys\ShopBundle\Model\Product`
+- classes in model must always represent a some kind of behavior of a model unit
+- model unit should always be represented by an Entity
+- model can be dependent on a component but not vice versa, this rule comes from definition of a [Component](../introduction/components.md).
+
+## What is and what is not a model
```

Comment:
Here I'm missing the mention of integration code not being a part of the Model namespace. Typically controllers, forms or grids (which are a part of Model atm but should be moved).

Otherwise, according to "That means that everything in a model should be related to some functionality of Domain, in our case, e-commerce." it should be a part of the Model namepsace. Eg. `ProductType` is related to the functionality of editing products.